sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -427,23 +447,21 @@ public WritableByteChannel create(GcsPath path, String type) throws IOException
    */
   public WritableByteChannel create(GcsPath path, String type, Integer uploadBufferSizeBytes)
       throws IOException {
-    GoogleCloudStorageWriteChannel channel =
-        new GoogleCloudStorageWriteChannel(
-            executorService,
-            storageClient,
-            new ClientRequestHelper<>(),
-            path.getBucket(),
-            path.getObject(),
-            type,
-            /* kmsKeyName= */ null,
-            AsyncWriteChannelOptions.newBuilder().build(),
-            new ObjectWriteConditions(),
-            Collections.emptyMap());
-    if (uploadBufferSizeBytes != null) {
-      channel.setUploadBufferSize(uploadBufferSizeBytes);
+    if (uploadBufferSizeBytes == null) {
+      return googleCloudStorage.create(new StorageResourceId(path.getBucket()));
```

Comment:
Can we combine this with code below? Does it need to call setGrpcEnabled, too?

Reply:
fixed.
I had to rework the code a bit to assume that gcsio doesn't yet have grpc support - but make this code almost ready to use it when it is available in gcsio

Reply:
fixed.
and uploaded the modifications.